frontend/pulse3d/src/components/interactiveAnalysis/WaveformGraph.js

```diff
@@ -883,7 +1117,33 @@ export default function WaveformGraph({
             left="5px"
             fontSize={15}
             borderRadius="5px"
-            clickFn={resetWellChanges}
+            clickFn={() => {
+              //set array here
+              let newArr = [...peakY1];
```

Comment:
This seems like the same 3 lines of code written over and over again, and I think I've seen these same 3 lines elsewhere. We should make a function for this as well

Reply:
This code in necessary for the state monitoring in react to recognize that the array has changed. Without this the state will not be updated and the limit lines will not be moved.

Reply:
The `updateArray` function I wrote in another comment can be used here